### ISSUE TITLE
Implement Event registry

### ISSUE THREAD

**Issue by Afforess** [Afforess/Factorio-Stdlib#4]
An alternate method of registering events is needed. It should allow multiple event handlers to be registered with one type of event, and allow the event to cascade through them, or be aborted. Example:

```
Event.register(defines.events.on_tick, function(event)
   ...
end).register(function(event)
 ...
end
```

_or_

```
Event.register(defines.events.on_tick, function(event)
   print("Hello World")
end)

Event.register(defines.events.on_tick, function(event)
  return true -- return true should cancel on_tick for any other event handlers for this event, so 'Goodbye!' should never be printed
end)

Event.register(defines.events.on_tick, function(event)
   print("Goodbye!")
end)
```

**Comment by Choumiko** [comment]
Since i'm messing with Events anyways right now, i made a rather crude implementation: https://github.com/Choumiko/Factorio-Stdlib/tree/events
Let me know what you think about it

**Comment by Afforess** [comment]
It's a solid start. I like it.

**Comment by Afforess** [comment]
Implemented in 0.3.0

**Comment by Choumiko** [comment]
Earlier today i noticed it is acting a bit different than script.on_event: https://github.com/Choumiko/Factorio-Stdlib/commit/43f9927c67117d8b3c94fad515df28d8a60431e6

I think something like Event.remove(event, handler) could be useful, it would remove the given handler from the event

Also stumbled upon this: [game.get_event_handler(event)](http://lua-api.factorio.com/0.12.30/LuaGameScript.html#get_event_handler) could also be added i think. Though i wonder why it is in game and not script?
